core/rfb.js

```diff
@@ -1069,18 +1074,46 @@ export default class RFB extends EventTargetMixin {
         return this._fail("No supported sub-auth types!");
     }
 
+    _negotiate_ultravnc_auth() {
+        if (this._sock.rQwait("UltraVNC auth result", 4)) {return false;}
+        this._sock.rQshift32();
+        this._rfb_init_state = 'Security';
```

Comment:
So we're going back to handshaking for authentication? Why can't we just pick the proper one right away?

Reply:
I guess not since you removed all of this?